gk/main.c

```diff
@@ -41,6 +329,17 @@ gk_proc(void *arg)
 
 	rte_atomic32_inc(&gk_conf->ref_cnt);
 
+	/* Setup the flow entry table using hash table. */
+	ret = setup_gk_instance(block_idx, gk_conf);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER, 
+			"gk: failed to setup gk instances for GK block at %u!\n",
+			lcore);
+		exiting = true;
```

Comment:
Indeed, if we put this setup function in the ``run_gk()``, we can avoid further initialization of other functional blocks. I will put the setup function for each gk instance within the ``for`` loop in ``run_gk()``.